src/channel.ts

```diff
@@ -1547,6 +1535,17 @@ export class Channel<StreamChatGenerics extends ExtendableGenerics = DefaultGene
         };
 
         if (read.user.id === user?.id) {
+          if (this.state.read[user.id].last_read) {
```

Comment:
Can't we only run this code if the underlying data has changed?

Reply:
Fixed and added tests